### PR TITLE
Use first Event as reference for timezone transitions

### PR DESCRIPTION
**Pull request by bauerandreas**
In Germany there where no daylight saving time in year 1970. So we need to reference the current year for getting time transitions to get right time offset.

### CODE REVIEW

**Comment by jasvrcek** [comment]
@bauerandreas you're right, this does need to be dealt with in some way, and using the earliest event in a calendar seems like a good compromise. I'm on board with that - can you fix the issue that is causing the tests to fail? (See the "Details" link next to the Travis CI build failed warning above. Here's an error output: https://travis-ci.org/jasvrcek/ICS/jobs/364514252.) Once that's done I'll merge.

**Comment by bauerandreas** [comment]
@jasvrcek great! I will fix the issue causing the tests to fail.

File: src/CalendarExport.php [hunk 1]
```diff
@@ -56,60 +56,72 @@ public function getStream()
...
-            $transitions = $tz->getTransitions(strtotime('1970-01-01'), strtotime('1970-12-31'));
-                
+
+            $calEvents = $cal->getEvents();
+
+            // Fallback to current year
+            $startYear = $endYear = date('Y');
+            if ($calEvents->first() !== false) {
+                // Take first event as referenece for timezone transitions
+                $firstEvent = $calEvents->manuallyAddedData[0];
```
Comment (felixkempf) [inline src/CalendarExport.php]: Use `first()` here too?

File: src/CalendarExport.php [hunk 2]
```diff
@@ -56,60 +56,72 @@ public function getStream()
...
             $tz = $cal->getTimezone();
-            $transitions = $tz->getTransitions(strtotime('1970-01-01'), strtotime('1970-12-31'));
-                
+
+            $calEvents = $cal->getEvents();
+
+            // Fallback to current year
+            $startYear = $endYear = date('Y');
+            if ($calEvents->first() !== false) {
+                // Take first event as referenece for timezone transitions
```
Comment (felixkempf) [inline src/CalendarExport.php]: typo

**Comment by jasvrcek** [comment]
Thanks for your work @bauerandreas and @felixkempf!

**Pull request merged**